stdlib/public/core/Array.swift

```diff
@@ -98,6 +98,18 @@
 ///     print(oddNumbers[0], oddNumbers[3], separator: ", ")
 ///     // Prints "1, 7"
 ///
+///     print(oddNumbers[-1])
+///     // Using a negative number
+///     // Triggers runtime error: Index out of range
+///
+///     print(oddNumbers[8])
+///     // Equal to `count`
+///     // Triggers runtime error: Index out of range
+///
+///     print(oddNumbers[9])
+///     // Greater than `count`
+///     // Triggers runtime error: Index out of range
+///
```

Comment:
Instead of adding a code example of every way to use an array incorrectly, maybe rewriting the paragraph above would help?  The prose already describes these invalid conditions, but it might be easier to read as a bulleted list.

Repeating the list from the paragraph in code makes the listing pretty long, and might actually make this harder to read.

Reply:
As a user of Swift, I would like to share my personal experience: I have found myself perplexed on occasions due to the lack of examples accompanying the explanations. In the case of Array, for instance, although the comments are neatly organized, the absence of detailed explanations means users may need to go through the trouble of referring to Apple's official documentation. If explanations were included, I believe it would not only facilitate a quicker understanding of the language but also offer easier accessibility, combining the benefits of both rapid comprehension and user-friendliness

Reply:
The comments that begin with `///` _are_ the documentation.  Any text we write here appears in the API reference for `Array`.  For example, the content this PR suggests changing is published here:

https://developer.apple.com/documentation/swift/array#Accessing-Array-Values

Reply:
Previously, there was an example about Arrays in the comments as well. I added a more detailed example to those comments with the intention of facilitating a better understanding. As I mentioned earlier, another advantage of this is that it can save the effort of consulting the official Apple documentation. That's why I included the additional example as described above.

Reply:
I would like to inquire whether it would be a good idea to delete the code from lines 101 to 112.

As a user, I often utilize the 'Jump to Definition' feature in Xcode when I want to learn about how to use a function or am curious about its internal implementation while developing an app or working on a project. When I use this feature, I find that the comments not only explain the internal workings but also provide detailed instructions on how to use the functions, allowing me to understand them thoroughly without needing to consult Apple's official documentation. Therefore, based on my experience, I suggest that adding examples to the existing comments explaining arrays in the Pull Request would make the descriptions longer but would significantly enhance the understanding and clarity of arrays. I believe this user-friendly approach would greatly benefit others.

cc: @amartini51